geode-core/src/main/java/org/apache/geode/internal/cache/GemFireCacheImpl.java

```diff
@@ -1086,6 +1088,27 @@ public static GemFireCacheImpl getForPdx(String reason) {
     clientMetadataService = clientMetadataServiceFactory.apply(this);
   }
 
+  @Override
+  public void lockDiskStore(String diskStoreName) {
+    CountDownLatch countDownLatch =
+        diskStoreLatches.putIfAbsent(diskStoreName, new CountDownLatch(1));
+    if (countDownLatch != null) {
+      try {
+        countDownLatch.await();
```

Comment:
`putIfAbsent` returns the old value (if any) or null. Are you trying to await() on the old Latch or the new one you just put in the ConcurrentMap?

Reply:
It will `await()` on the old latch. More specifically, the first thread will put a new `CountDownLatch` in the map. The other threads will `await()` on the latch that the first thread put in the map.